### DISCUSSION TITLE
please how to disable auto indentation and auto code editing

### DISCUSSION THREAD

**Discussion by Zorono** [microsoft/vscode-cpptools#12457] [Q&A]
please how to disable auto indentation and auto code editing ??
like 
```
		Callback(AMX *amx, int cb_idx, ParamList_t &&params) :
			_amx(amx),
			_amx_index(cb_idx),
			_params(std::move(params))
		{ }
```
is changed to
```
		Callback(AMX *amx, int cb_idx, ParamList_t &&params) : _amx(amx),
												       _amx_index(cb_idx),
												       _params(std::move(params))
		{
		}
```

**Comment by Yewzir** [comment]
@Zorono 

Go to your **settings** in Visual Studio Code, and perform a search based on the keyword _formatter_. You may want to turn the defaults off.

![image](https://github.com/user-attachments/assets/e72b0206-d87a-4ebd-b404-861f558efc73)

Next, perform a search based on the keyword _indent_. **Auto Indent** is just what you need, or rather don't need :wink:

![image](https://github.com/user-attachments/assets/1006fe37-cade-45d3-b924-ce808397b79c)

Additionally, it may be worth it to browse through all of the **Formatting** settings of the C/C++ extension, there is a significant amount of settings to adjust eventually. Overriding defaults should be enough, however.

![image](https://github.com/user-attachments/assets/8815f286-9352-43a7-b0eb-1b9e5089a342)

Kind regards
Yewzir
